### DISCUSSION TITLE
XBOX Elite Series 2 Controller Not Detected

### DISCUSSION THREAD

**Discussion by michaeldpotter** [unknownskl/xbox-xcloud-client#1060] [General]
Hello,
I'm running Greenlight 2.0.0-beta15 on macOS Sonoma 14.1.2

I'm able to pair a normal XBOX Series X/S controller with my Mac, and then in settings in Greenlight, it is detected. However, when I try the same thing with my XBOX Elite Series 2 controller, it pairs fine with the Mac, but is not detected in Greenlight.

Has anyone ran into this issue or have any ideas on how to resolve it or troubleshoot it?

Thanks!

**Comment by idlehand79** [comment]
Only way I have got it to work is to run it via terminal "sudo /Applications/Greenlight.app/Contents/MacOS/Greenlight
"

**Reply by michaeldpotter** [reply]
This worked for me as well! Thank you so much for posting this work around! 💯

**Reply by ktiedt** [reply]
6 months later, how is this still not a more readily discussed issue (or fixed) Luckily Google indexed this because I just spent 40 mins trying to figure out why only one application (Greenlight) couldnt see my controller.

**Reply by c9stealth** [reply]
did this fix your issues Ktiedt?  it doesnt' seem to be working for me.

**Reply by ktiedt** [reply]
> did this fix your issues Ktiedt?  it doesnt' seem to be working for me. 

Yup it worked for me, haven't used it as much sense but I don't recall having to use this since that day for it to work.

**Comment by Shusanta** [comment]
still no luck here

**Comment by N1k0la1** [comment]
Knowing this, I created a launcher with automater. 

1. First create a shell script in Terminal that contains the command above:

`sudo nano /usr/local/bin/launch_greenlight.sh`

2. Add this code:

`#!/bin/bash`
`/Applications/Greenlight.app/Contents/MacOS/Greenlight`

3. Then make the shell script executable:
`sudo chmod +x /usr/local/bin/launch_greenlight.sh`

4. (Optional) Add the Script to sudoers so you don't have to enter your password everytime you start the script:
`sudo visudo`

5.(Optional) Add this to the end (make sure to update "Your_Username" with your MacOS username. (use `whoami` in Terminal to get your username):
`YOUR_USERNAME ALL=(ALL) NOPASSWD: /usr/local/bin/launch_greenlight.sh`

6. Now start Automater and choose Application as type.
7. Search for "Run Shell Script" and double click the result
8. Now replace `cat` with this line:
9. `/usr/local/bin/launch_greenlight.sh`
10. Save the file to the Applications folder
11. (Optional) If you want to edit the Icon, choose the image you want to use (.png or .icns), resize it to 512x512 and save.
12. (Optional) Then go to the Applications folder and right-click your created app (that has the automater logo) and click "Get Info".
13. (Optional) Now drag your resized image onto the small icon in the top left of the Info window.
14. Done

Tested this on MacOS Sequoia 15.1 and works like a charm.

**Reply by michaeldpotter** [reply]
Thanks for this great write-up @N1k0la1!

I think the only edit I would make is not to directly edit the /etc/sudoers file with visudo, but to instead create a new file in /private/etc/sudoers.d This way system updates won't potentially overwrite the changes in /etc/sudoers.

Thanks again!

**Comment by Tylerstonge69** [comment]
I'm completely lost on how to get that sudo to run from what I see a normal controller works without this?

**Reply by michaeldpotter** [reply]
I'd start troubleshooting this by opening a terminal and running this command:

`sudo /Applications/Greenlight.app/Contents/MacOS/Greenlight`

Let us know the results from that.

I have been able to get a non-elite controller to work fine without having to run it via sudo.

**Comment by Tylerstonge69** [comment]
I got it working in Safari with a web portal
I’m kinda thinking it’s something apple is not allowing the app to see a
game pad
